spring-pulsar/src/main/java/org/springframework/pulsar/annotation/ReactivePulsarListenerAnnotationBeanPostProcessor.java

```diff
@@ -0,0 +1,785 @@
+import org.springframework.core.convert.TypeDescriptor;
+import org.springframework.core.convert.converter.ConditionalGenericConverter;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.core.convert.converter.GenericConverter;
+import org.springframework.core.log.LogAccessor;
+import org.springframework.format.Formatter;
+import org.springframework.format.FormatterRegistry;
+import org.springframework.format.support.DefaultFormattingConversionService;
+import org.springframework.lang.Nullable;
+import org.springframework.messaging.converter.GenericMessageConverter;
+import org.springframework.messaging.handler.annotation.support.DefaultMessageHandlerMethodFactory;
+import org.springframework.messaging.handler.annotation.support.MessageHandlerMethodFactory;
+import org.springframework.messaging.handler.invocation.HandlerMethodArgumentResolver;
+import org.springframework.messaging.handler.invocation.InvocableHandlerMethod;
+import org.springframework.pulsar.config.PulsarListenerBeanNames;
+import org.springframework.pulsar.config.PulsarListenerEndpointRegistrar;
+import org.springframework.pulsar.config.reactive.MethodReactivePulsarListenerEndpoint;
+import org.springframework.pulsar.config.reactive.ReactivePulsarListenerContainerFactory;
+import org.springframework.pulsar.config.reactive.ReactivePulsarListenerEndpoint;
+import org.springframework.pulsar.config.reactive.ReactivePulsarListenerEndpointRegistry;
+import org.springframework.pulsar.core.reactive.ReactiveMessageConsumerBuilderCustomizer;
+import org.springframework.util.Assert;
+import org.springframework.util.ReflectionUtils;
+import org.springframework.util.StringUtils;
+import org.springframework.validation.Validator;
+
+/**
+ * Bean post-processor that registers methods annotated with
+ * {@link ReactivePulsarListener} to be invoked by a Pulsar message listener container
+ * created under the covers by a {@link ReactivePulsarListenerContainerFactory} according
+ * to the parameters of the annotation.
+ *
+ * <p>
+ * Annotated methods can use flexible arguments as defined by
+ * {@link ReactivePulsarListener}.
+ *
+ * <p>
+ * This post-processor is automatically registered by the {@link EnablePulsar} annotation.
+ *
+ * <p>
+ * Auto-detect any {@link PulsarListenerConfigurer} instances in the container, allowing
+ * for customization of the registry to be used, the default container factory or for
+ * fine-grained control over endpoints registration. See {@link EnablePulsar} Javadoc for
+ * complete usage details.
+ *
+ * @param <V> the payload type.
+ * @author Christophe Bornet
+ * @see ReactivePulsarListener
+ * @see EnablePulsar
+ * @see PulsarListenerConfigurer
+ * @see PulsarListenerEndpointRegistrar
+ * @see ReactivePulsarListenerEndpointRegistry
+ * @see ReactivePulsarListenerEndpoint
+ * @see MethodReactivePulsarListenerEndpoint
+ */
+public class ReactivePulsarListenerAnnotationBeanPostProcessor<V>
```

Comment:
A lot of this should be mutualized with `PulsarListenerAnnotationBeanPostProcessor` I guess.

Reply:
Sounds like a good idea.

Reply:
Agreed. Once the "dust settles" from this PR lets come back and see what makes sense to harmonize.